trlx/trainer/nemo_ppo_trainer.py

```diff
@@ -0,0 +1,323 @@
+from logging import getLogger
+from math import ceil
+from pathlib import Path
+from typing import Any, Callable, Iterable, Iterator, List, Optional, Tuple, Union, cast
+
+import torch
+import transformers
+import wandb
+from apex.transformer import parallel_state
+from nemo.utils import logging
+from omegaconf.omegaconf import OmegaConf
+from torch.utils.data import DataLoader, DistributedSampler
+from tqdm import tqdm
+
+from trlx.data.accelerate_base_datatypes import PromptBatch
+from trlx.data.configs import TRLConfig
+from trlx.data.ilql_types import flatten_dataclass
+from trlx.data.ppo_types import PPORLBatch, PPORLElement
+from trlx.models.modeling_nemo_ppo import PPOGPT
+from trlx.models.modeling_ppo import AdaptiveKLController, FixedKLController, PPOConfig
+from trlx.pipeline.offline_pipeline import PromptPipeline
+from trlx.pipeline.ppo_pipeline import ppo_collate_fn
+from trlx.trainer import BaseRLTrainer, register_trainer
+from trlx.trainer.nemo_ilql_trainer import ShuffledCyclicSequence, megatron_trainer
+from trlx.utils import infinite_dataloader
+from trlx.utils.modeling import logprobs_of_labels
+
+logging = getLogger(__name__)
+
+
+@register_trainer
+class NeMoPPOTrainer(BaseRLTrainer):
+    def __init__(
+        self,
+        config: TRLConfig,
+        metric_fn: Optional[Callable[[List[str]], Any]] = None,
+        megatron_cfg: Optional[Union[str, dict]] = None,
+        pretrained_model: Optional[str] = None,
+        **kwargs,
+    ):
+        super().__init__(config, metric_fn=metric_fn, **kwargs)
+
+        if metric_fn is None:
+
+            def metric_fn(*args, **kwargs):
```

Comment:
honestly we should print a warning if someone doesn't set the metric function.

Reply:
for accelerate we automatically set it to the reward fn too